lib/oauth2client.js

```diff
@@ -263,6 +291,85 @@ OAuth2Client.prototype._validateTokenResponse = function(body) {
 };
 
 /**
+ * Validates the response returned from an OAuth 2.0 device code request.
+ * @private
+ * @param  {string} body  The response as a string encoded JSON object.
+ * @return {object}       The parsed response.
+ */
+OAuth2Client.prototype._validateDeviceCodeResponse = function(body) {
+   var wireResponse;
+   var deviceCodeResponse = {};
+
+   try {
+      wireResponse = JSON.parse(body);
+   } catch(e) {
+      throw new Error('The device code response returned from the server is unparseable as JSON.');
+   }
+
+   var intKeys = [
+        DeviceCodeResponseParameters.EXPIRES_IN, 
+        DeviceCodeResponseParameters.INTERVAL
+   ];
+
+   this._parseOptionalInts(wireResponse, intKeys);
+  
+   if (!wireResponse[DeviceCodeResponseParameters.EXPIRES_IN]){
+      throw this._log.createError('wireResponse is missing expires_in');
+   }
+
+   if (!wireResponse[DeviceCodeResponseParameters.DEVICE_CODE]) {
+      throw this._log.createError('wireResponse is missing device code');
+   }
+
+   if (!wireResponse[DeviceCodeResponseParameters.USER_CODE]) {
+      throw this._log.createError('wireResponse is missing user code');
+   }
+
+   mapFields(wireResponse, deviceCodeResponse, DEVICE_CODE_RESPONSE_MAP);
+
+   return deviceCodeResponse;
+};
+
+/**
+ * @private
+ * @param {string}                   body        The body of a http token response.
+ * @param {OAuth2Client.GetTokenCallback}    callback    A call back function.  The body parameter is the body parameter passed
+ *                                               into this function.
+ */
+OAuth2Client.prototype._handlePollingResponse = function(body) {
+    //handle token error response
+    var tokenResponse = this._handlePollingRequestErrorResponse(body);
+    if (_.isEmpty(tokenResponse)){
+       tokenResponse = this._validateTokenResponse(body);
+    }
+
+    return tokenResponse;
+};
+
+/**
+ * @private
+ * @param {string}                   body        The body of a http token response.
+ * @param {OAuth2Client.GetTokenCallback}    callback    A call back function.  The body parameter is the body parameter passed
```

Comment:
This method currently does not accept a callback. We should either add a callback parameter or remove the @param tag.